shared/actions/settings.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import * as Constants from '../constants/settings'
+import {apiserverGetRpcPromise} from '../constants/types/flow-types'
+import {call, put} from 'redux-saga/effects'
+import {takeLatest} from 'redux-saga'
+
+import type {SagaGenerator} from '../constants/types/saga'
+import type {NotificationsRefresh} from '../constants/settings'
+
+function notificationsRefresh (): NotificationsRefresh {
+  return {type: Constants.notificationsRefresh, payload: undefined}
+}
+
+function * refreshNotificationsSaga (): SagaGenerator<any, any> {
+  try {
+    const json: ?{body: string} = yield call(apiserverGetRpcPromise, {
+      param: {
+        endpoint: 'account/subscriptions',
+        args: [],
+      },
+    })
+
+    const results: {
+      notifications: {
+        email: {
+          settings: Array<{
+            name: string,
+            description: string,
+            subscribed: boolean,
+          }>,
+          unsub: boolean,
+        },
+      },
+    } = JSON.parse(json && json.body || '')
+
+    const unsubscribedFromAll = results.notifications.email.unsub
+
+    const settings = results.notifications.email.settings.map(s => ({
+      name: s.name,
+      subscribed: s.subscribed,
+      description: s.description,
+    })) || []
+
+    yield put({
+      type: Constants.notificationsRefreshed,
+      payload: {
+        unsubscribedFromAll,
+        settings,
+      },
+    })
+  } catch (err) {
+    // TODO global error
```

Comment:
Maybe we should remove the `try...catch` for now so that it doesn't swallow errors?